### PR TITLE
Add VHLO support for FP8

### LINKED ISSUE #906: Add VHLO support for FP8
### Request description

Recently we added FP8 to StableHLO. This needs to be supported by VLHO for compatibility guarantees.

### PR DESCRIPTION
**Pull request by GleasonK**
This provides compatibility guarantees for FP8 types `f8E4M3FN` and `f8E5M2`.

This is built on #849, use that as a diffbase. This change is relatively simple and shows all that goes into supporting a new datatype in VHLO.

Closes #906.

### CODE REVIEW

File: stablehlo/dialect/VhloBytecode.cpp [hunk 1]
```diff
@@ -321,6 +321,16 @@ enum TypeCode {
   ///   WitnessType {
   ///   }
   kWitnessType = 18,
+
+  ///   Float8E4M3FN {
+  ///   }
+  ///
```
Comment (burmako) [inline stablehlo/dialect/VhloBytecode.cpp:327]: Extraneous newline?

File: stablehlo/dialect/VhloBytecode.cpp [hunk 2]
```diff
@@ -1123,6 +1131,10 @@ Type VhloBytecodeInterface::readType(DialectBytecodeReader &reader) const {
       return BFloat16V1Type::get(getContext());
     case vhlo_encoding::kComplexType:
       return readComplexType(reader);
+    case vhlo_encoding::kFloat8E5M2:
```
Comment (burmako) [inline stablehlo/dialect/VhloBytecode.cpp:1134]: Let's order alphabetically? (Same on several occasions below).

File: stablehlo/dialect/VhloBytecode.cpp [hunk 3]
```diff
@@ -321,6 +321,16 @@ enum TypeCode {
   ///   WitnessType {
```
Comment (burmako) [inline stablehlo/dialect/VhloBytecode.cpp:321]: Is this PR a good time to update VhloDialect.td to call out the addition of fp8 support to 0.4.0?

**Review by burmako — CHANGES_REQUESTED** [review]

File: stablehlo/dialect/VhloBytecode.cpp (hunk 3 above)
Comment (GleasonK) [reply stablehlo/dialect/VhloBytecode.cpp:321]: Yes. Good catch. Will need to rebase on main to avoid conflicts.

**Review by burmako — APPROVED** [review]

**Pull request merged**